components/tls/tls-mpc/src/follower.rs

```diff
@@ -182,6 +190,71 @@ impl MpcTlsFollower {
         Ok(())
     }
 
+    async fn handle_encrypt_msg(&mut self, msg: EncryptMessage) -> Result<(), MpcTlsError> {
+        let EncryptMessage { typ, seq, len } = msg;
+
+        if self.total_bytes_transferred() + len > self.config.common().max_transcript_size() {
+            return Err(MpcTlsError::MaxTranscriptLengthExceeded(
+                self.total_bytes_transferred() + len,
+                self.config.common().max_transcript_size(),
+            ));
+        }
+
+        self.encrypter.encrypt_blind(typ, seq, len).await
+    }
+
+    async fn handle_decrypt_msg(&mut self, msg: DecryptMessage) -> Result<(), MpcTlsError> {
+        let DecryptMessage {
+            typ,
+            explicit_nonce,
+            ciphertext,
+            seq,
+        } = msg;
+
+        if self.total_bytes_transferred() + ciphertext.len()
+            > self.config.common().max_transcript_size()
+        {
+            return Err(MpcTlsError::MaxTranscriptLengthExceeded(
+                self.total_bytes_transferred() + ciphertext.len(),
+                self.config.common().max_transcript_size(),
+            ));
+        }
+
+        match typ {
+            ContentType::ApplicationData => {
+                self.decrypter
+                    .decrypt_blind(typ, explicit_nonce, ciphertext, seq)
+                    .await
+            }
+            ContentType::Alert => {
+                let bytes = self
+                    .decrypter
+                    .decrypt_public(typ, explicit_nonce, ciphertext, seq)
+                    .await?;
+
+                let alert = AlertMessagePayload::read_bytes(&bytes)
+                    .ok_or(MpcTlsError::PayloadDecodingError)?;
+
+                if alert.level == AlertLevel::Fatal {
+                    return Err(MpcTlsError::ReceivedFatalAlert);
+                }
+
+                if alert.description == AlertDescription::CloseNotify {
+                    self.closed = true;
+                }
+
+                Ok(())
+            }
+            typ => Err(MpcTlsError::UnexpectedContentType(typ)),
+        }
+    }
+
+    async fn handle_close_notify(&mut self, msg: EncryptMessage) -> Result<(), MpcTlsError> {
+        let EncryptMessage { typ, seq, len } = msg;
+
+        self.encrypter.encrypt_blind(typ, seq, len).await
```

Comment:
The reason why we don't use `encrypt_public` here is because it is not implemented in `follower::Encrypter`, correct?
if so, we can leave a TODO comment to convey that it is not strictly required to use  `encrypt_blind` here.